Deadlock.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeReviewDemo
+{
+    internal class DeadlockDemo
+    {
+        object myLock1 = new object();
+        object myLock2 = new object();
+        void StartThreads()
+        {
```

Comment:
Ensure that `Thread.Sleep(1000);` is necessary for your logic. If it's for demonstration purposes only, consider adding a comment to explain its usage. Unnecessary sleeps can lead to performance issues in production code.